pcli/src/opt.rs

```diff
@@ -88,7 +86,13 @@ impl Opt {
         let fvk = wallet.spend_key.full_viewing_key().clone();
 
         // ...and the view service...
-        let view = self.view_client(&fvk).await?;
+        // There is one command that is both offline and *un*synced:
+        // `pcli view balance --offline`. Hence, this check.
+        let view = if !self.cmd.offline() || matches!(self.cmd, Command::View(Balance(_))) {
+            Some(self.view_client(&fvk).await?)
+        } else {
+            None
+        };
```

Comment:
As above, I think the simpler thing here would just be to remove offline balance support. As noted in your comment here, it's a bit of a weird outlier.